NanoGames.Core/Vector.cs

```diff
@@ -97,6 +102,17 @@ public Vector(double x, double y)
             return !(a == b);
         }
 
+        public static bool operator <(Vector a, Vector b)
```

Comment:
This is rather counter-intuitive - commonly, a vector is considered "smaller" than another vector if its length is smaller. Overloaded operators should behave intuitively, see also https://docs.microsoft.com/de-de/cpp/cpp/general-rules-for-operator-overloading
Thus, I suggest to convert this to a method.